tests/lib/hardware/hardware_base.py

```diff
@@ -151,15 +151,22 @@ def ansible_run_playbook(self, playbook: str,
         else:
             limit = ""
 
+        extra_vars_files=""
         # Supplied extra_vars from settings always take precedence
         if settings.ANSIBLE_EXTRA_VARS:
-            extra_vars.update(json.loads(settings.ANSIBLE_EXTRA_VARS))
```

Comment:
I wonder if we should change the method parameter `extra_vars` to just expect a string which can either be json or a string starting with `@` or even something like `version=1.23.45` (see https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#defining-variables-at-runtime fro all possibilities). Then we could just do `--extra-vars {extra_vars}`

Reply:
I think we should decouple the method parameter `extra_vars`, from the settings variable `ANSIBLE_EXTRA_VARS`.

Because we can apply multiple `--extra-vars` we should just supply both (`--extra-vars {extra_vars} --extra-vars {settings.ANSIBLE_EXTRA_VARS}`) giving both the option to be in any format supported by ansible.

Reply:
I love you are discussing this here, but could you please merge this PR to unblock me, and create another PR to with your suggestion, I am happy you guys want to replicate all the features of ansible in rookcheck, but I don't like you force me to do this in this PR.
My intention in this patch was not to break existing functionality in rookcheck to not interfere someone else work.

Regarding @toabctl comment if you want to just have something like `ROOKCHECK_ANSIBLE_EXTRA_VARS=--extra-vars {extra_vars}`, it would not go with this, it can be confusing, I would suggest to use `ROOKCHECK_ANSIBLE_EXTRA_ARGS` for that purpose and we can drop the ROOKCHECK__ANSIBLE_EXTRA_VARS since it is useless.

Regardging @jhesketh comment in the context of rookcheck I don't see any advantage of using multipe --extra-vars when passing the parameters to ansible via tox|pytest, I would prefer to have a **single** file do not pack every parameter to the command line.

Reply:
> I love you are discussing this here, but could you please merge this PR to unblock me, and create another PR to with your suggestion, I am happy you guys want to replicate all the features of ansible in rookcheck, but I don't like you force me to do this in this PR.

Not looking to re-implement Ansible. My suggestions were for a simplifier and more complete solution.

> My intention in this patch was not to break existing functionality in rookcheck to not interfere someone else work.

The suggested changes should not break existing functionality.

> 
> Regarding @toabctl comment if you want to just have something like `ROOKCHECK_ANSIBLE_EXTRA_VARS=--extra-vars {extra_vars}`, it would not go with this, it can be confusing, I would suggest to use `ROOKCHECK_ANSIBLE_EXTRA_ARGS` for that purpose and we can drop the ROOKCHECK__ANSIBLE_EXTRA_VARS since it is useless.
> 
> Regardging @jhesketh comment in the context of rookcheck I don't see any advantage of using multipe --extra-vars when passing the parameters to ansible via tox|pytest, I would prefer to have a **single** file do not pack every parameter to the command line.

The multiple `--extra-vars` isn't for having to pack each variable as a parameter. It is because we have extra vars coming from two places. 1) some tests and internals of rookcheck may need to overwrite variables at different times, and 2) the user may want to overwrite variables in configuration.

My suggestion simplifies the implementation by separating the two sources of vars, and passing them in directly giving both the test writer and the configuring user the option to use files, json, or otherwise (as supported by Ansible).

Hope that makes more sense, but let me know if not.

Reply:
> > My intention in this patch was not to break existing functionality in rookcheck to not int> > Regarding @toabctl comment if you want to just have something like `ROOKCHECK_ANSIBLE_EXTRA_VARS=--extra-vars {extra_vars}`, it would not go with this, it can be confusing, I would suggest to use `ROOKCHECK_ANSIBLE_EXTRA_ARGS` for that purpose and we can drop the ROOKCHECK__ANSIBLE_EXTRA_VARS since it is useless.
> > Regardging @jhesketh comment in the context of rookcheck I don't see any advantage of using multipe --extra-vars when passing the parameters to ansible via tox|pytest, I would prefer to have a **single** file do not pack every parameter to the command line.
> 
erfere someone else work.
> 
> The suggested changes should not break existing functionality.
> 

To be honest I don't understand the changes you suggest. Looking forward to review your upcoming PR.

> The multiple `--extra-vars` isn't for having to pack each variable as a parameter. It is because we have extra vars coming from two places. 1) some tests and internals of rookcheck may need to overwrite variables at different times, and 2) the user may want to overwrite variables in configuration.

I didn't see that in the existing code, and what the examples say it packs json into the variable. This is why this PR appeared.

> 
> My suggestion simplifies the implementation by separating the two sources of vars, and passing them in directly giving both the test writer and the configuring user the option to use files, json, or otherwise (as supported by Ansible).
> 
> Hope that makes more sense, but let me know if not.

I understand the sense of original ansible-playbook options very well. What I do not understand is your suggestion, it would be easier if you provided code examples. Anyways we are discussing the code which is not created by me, I was not going to change it, and more the current PR revision does not even touch it.

I hope we merge it asap, so I do not need to cherry-pick it from branch to branch in order to use the functionality.

Reply:
I think the idea from @jhesketh is https://github.com/SUSE/rookcheck/pull/216

Reply:
@toabctl That looks like reverting his previous change df75e6c

Reply:
No (unless I'm missing something). My previous change updated the vars by loading in the settings as a python dict and updating them then passing a final version of the dict to Ansible by one `--extra-vars`. #216 passes the settings as a second `--extra-vars` when once applied will overwrite anything else in the same order of precedence.

Reply:
@jhesketh I might be wrong but after Tom updated his initial version of the PR it is not now.